deployment/src/strongmind_deployment/container.py

```diff
@@ -30,7 +30,7 @@ def __init__(self, name, opts=None, **kwargs):
         self.memory = kwargs.get("memory", 512)
         self.entry_point = kwargs.get('entry_point')
         self.env_vars = kwargs.get('env_vars', {})
-        self.secrets = []
+        self.secrets = kwargs.get('secrets', [])
```

Comment:
I think I'd rather have the secrets component be part of the container component rather than the rails one. Because we may have other non-rails containers that need secrets.

Reply:
> I think I'd rather have the secrets component be part of the container component rather than the rails one. Because we may have other non-rails containers that need secrets.

We want to have 1 secret with all credentials per env (dev, stage, prod)
I went to 1 layer down to create SecretComponent inside RailsConponent and pass the credentials its contain to ContainerComponent
Because we can have several ContainerComponent Pulumi will try to create SecretComponent each time it creates ContainerComponent

Reply:
I see. Makes sense.
The only thing we lose, I suppose, with this approach, is that a ContainerComponent won't have secrets manager out of the box. But it can be added pretty easily. If we decide to make that built in later we can provide a way to opt-in (or opt-out when doing rails).

Reply:
Sure